File: src/net/sourceforge/reb4j/Flag.java
package net.sourceforge.reb4j;

/**
 * This enumeration represents flags that may be embedded into regular 
 * expressions.
 * 
 * @author Richard W. Eggert II
 * @see {@link Regex#enable(Flag...)}
 * @see {@link Regex#disable(Flag...)}
 * @see {@link Regex#enableDirective(Flag...)}
 * @see {@link Regex#disableDirective(Flag...)}
 *
 */
public enum Flag
{
	CASE_INSENSITIVE, // 'i'
	UNIX_LINES, // 'd'
	MULTILINE, // 'm'
	DOTALL, // 's'
	UNICODE_CASE, // 'u' 
	COMMENTS // 'x'
}

File: src/main/scala/net/sourceforge/reb4j/scala/Flag.scala
package net.sourceforge.reb4j.scala

/**
 * Flag that can be passed to the {@link java.util.regex.Pattern} 
 * regular expression engine.
 */
@SerialVersionUID(1L)
sealed class Flag private (val c : Char) extends Serializable 
	with NotNull with Immutable
{
	override final def toString = String.valueOf(c)
}


object Flag
{
	case object CaseInsensitive extends Flag('i')
	case object UnixLines extends Flag('d')
	case object Multiline extends Flag('m')
	case object DotAll extends Flag('s')
	case object UnicodeCase extends Flag('u') 
	case object Comments extends Flag('x')
	
	/**
	 * Wraps the specified expression in a group that enables the specified 
	 * matcher flags.
	 */
	def enable(nested : Expression, flags : Flag*) = Group.EnableFlags(nested, flags : _*) 
	
	/**
	 * Wraps the specified expression in a group that disables the specified
	 * matcher flags.
	 */
	def disable(nested : Expression, flags : Flag*) = Group.DisableFlags(nested, flags : _*)
	
}

